lib/index.js

```diff
@@ -185,33 +188,43 @@ MapboxGeocoder.prototype = {
     this._eventEmitter.emit('loading', { query: searchInput });
 
     // Possible config proprerties to pass to client
-    var keys = ['bbox', 'limit', 'countries', 'types', 'language', 'reverseMode'];
+    var keys = [
+      'bbox',
+      'limit',
+      'countries',
+      'types',
+      'language',
+      'reverseMode'
+    ];
     var self = this;
     // Create config object
     var config = keys.reduce(function(config, key) {
       if (self.options[key]) {
         // countries, types, and language need to be passed in as arrays to client
         // https://github.com/mapbox/mapbox-sdk-js/blob/master/services/geocoding.js#L38-L47
-        ['countries', 'types', 'language'].indexOf(key) > -1 ?
-          config[key] = [self.options[key]]
-        : config[key] = self.options[key]
+        ['countries', 'types', 'language'].indexOf(key) > -1
+          ? (config[key] = self.options[key].split(', '))
+          : (config[key] = self.options[key]);
       }
       return config;
     }, {});
 
     var request;
     // check if searchInput resembles coordinates, and if it does,
     // make the request a reverseGeocode
-    if (/^(\-?\d+(\.\d+)?),\s*(\-?\d+(\.\d+)?)$/.test(searchInput)) {
+    if (
+      this.options.reverseGeocode &&
+      /(-?\d+\.?\d*)[, ]+(-?\d+\.?\d*)[ ]*$/.test(searchInput)
+    ) {
       // parse coordinates
       var coords = searchInput.split(',').map(function(c) {
```

Comment:
This breaks down if I query with a space since the regex accepts it but this splitting code doesn't.

eg. query `10 10`

Reply:
Fixed in latest commit